pkg/validate/container.go

```diff
@@ -201,6 +203,117 @@ var _ = framework.KubeDescribe("Container", func() {
+		})
+
+		It("mount with 'rshared' should support propagation from host to container and vice versa", func() {
+			By("create host path and flag file")
+			mntSource, propagationSrcDir, propagationMntPoint, clearHostPath := createHostPathForMountPropagation(podID, runtimeapi.MountPropagation_PROPAGATION_BIDIRECTIONAL)
+			defer clearHostPath() // clean up the TempDir
+
+			By("create container with volume")
+			containerID := createMountPropagationContainer(rc, ic, "mount-propagation-test-", podID, podConfig, mntSource, runtimeapi.MountPropagation_PROPAGATION_BIDIRECTIONAL)
+
+			By("test start container with volume")
+			testStartContainer(rc, containerID)
+
+			By("create a propatation mount point in host")
+			createPropagationMountPoint(propagationSrcDir, propagationMntPoint)
+
+			By("check whether propagationMntPoint contains file or dir in container")
+			command := []string{"ls", "-A", propagationMntPoint}
+			output := execSyncContainer(rc, containerID, command)
+			Expect(len(output)).NotTo(BeZero(), "len(output) should not be zero.")
+
+			By("create a directory named containerMntPoint as a mount point in container")
+			containerMntPoint := path.Join(mntSource, "containerMntPoint")
+			command = []string{"sh", "-c", "mkdir -p " + containerMntPoint}
+			execSyncContainer(rc, containerID, command)
+
+			By("mount /etc to the mount point in container")
+			command = []string{"sh", "-c", "mount --bind /etc " + containerMntPoint}
+			execSyncContainer(rc, containerID, command)
+
+			By("check whether containerMntPoint contains file or dir in host")
+			fileInfo, err := ioutil.ReadDir(containerMntPoint)
+			framework.ExpectNoError(err, "failed to ReadDir %q in Host", containerMntPoint)
+			Expect(len(fileInfo)).NotTo(BeZero(), "len(fileInfo) should not be zero.")
+		})
+
+		It("mount with 'rslave' should support propagation from host to container", func() {
+			By("create host path and flag file")
+			mntSource, propagationSrcDir, propagationMntPoint, clearHostPath := createHostPathForMountPropagation(podID, runtimeapi.MountPropagation_PROPAGATION_BIDIRECTIONAL)
+			defer clearHostPath() // clean up the TempDir
+
+			By("create container with volume")
+			containerID := createMountPropagationContainer(rc, ic, "mount-propagation-test-", podID, podConfig, mntSource, runtimeapi.MountPropagation_PROPAGATION_BIDIRECTIONAL)
+
+			By("test start container with volume")
+			testStartContainer(rc, containerID)
+
+			By("create a propatation mount point in host")
+			createPropagationMountPoint(propagationSrcDir, propagationMntPoint)
+
+			By("check whether propagationMntPoint contains file or dir in container")
+			command := []string{"ls", "-A", propagationMntPoint}
+			output := execSyncContainer(rc, containerID, command)
+			Expect(len(output)).NotTo(BeZero(), "len(output) should not be zero.")
+		})
+
+		It("mount without propagation option should not support propagation", func() {
```

Comment:
This is not defined in CRI, runtime may return an error for non-supported mount propagation option, so let's not test this.

Reply:
OK, will remove this part.